File: day2/day2part2.py
guide = open("/home/kat/AOC2022/day2/input", "r")
myScore = 0
badScore = 0

for line in guide.readlines():
  moves = line.split()
  badMove = moves[0]
  myMove = moves[1]
  match badMove:
    case "A":
      match myMove:
        case "X":
          myScore = myScore + 3
          badScore = badScore + 7 
        case "Y":
          myScore = myScore + 4
          badScore = badScore + 4
        case "Z":
          myScore = myScore + 8
          badScore = badScore + 1
    case "B":
      match myMove:
        case "X":
          myScore = myScore + 1
          badScore = badScore + 8
        case "Y":
          myScore = myScore + 5
          badScore = badScore + 5
        case "Z":
          myScore = myScore + 9
          badScore = badScore + 2
    case "C":
      match myMove:
        case "X":
          myScore = myScore + 2
          badScore = badScore + 9
        case "Y":
          myScore = myScore + 6
          badScore = badScore + 6
        case "Z":
          myScore = myScore + 7
          badScore = badScore + 3

print("My Score: ",myScore)
print("Their Score: ",badScore)

File: day2/day2part2.rs
use std::fs::File;
use std::io::{BufRead, BufReader, Error};

fn main() -> Result<(), Error> {
    let path = "/home/kat/AOC2022/day2/input";
    let input = BufReader::new(File::open(path)?);
    let mut my_score = 0;
    let mut bad_score = 0;

    for line in input.lines() {
        match line {
            Ok(line) => {
                let mut moves = line.chars();
                let bad_move = moves.nth(0).unwrap();
                let my_move = moves.nth(1).unwrap();
                match bad_move{
                    'A'=> {
                        match my_move {
                            'X'=> {
                                my_score += 3;
                                bad_score += 7;
                            }
                            'Y' => {
                                my_score += 4;
                                bad_score += 4;
                            }
                            'Z' => {
                                my_score += 8;
                                bad_score += 1;
                            }
                            _ => {}
                        }
                    }
                    'B'=> {
                        match my_move {
                            'X' => {
                                my_score += 1;
                                bad_score += 8;
                            }
                            'Y' => {
                                my_score += 5;
                                bad_score += 5;
                            }
                            'Z' => {
                                my_score += 9;
                                bad_score += 2;
                            }
                            _ => {}
                        }
                    }
                    'C'=> {
                        match my_move {
                            'X' => {
                                my_score += 2;
                                bad_score += 9;
                            }
                            'Y' => {
                                my_score += 6;
                                bad_score += 6;
                            }
                            'Z' => {
                                my_score += 7;
                                bad_score += 3;
                            }
                            _ => {}
                        }
                    }
                    _ => {}
                }

            }
            Err(e) => {
                return Err(e);
            }
        }
    }
    println!("My Score: {}", my_score);
    println!("Their Score: {}", bad_score);
    Ok(())
}